crates/anvil-polkadot/src/substrate_node/service.rs

```diff
@@ -97,6 +94,12 @@ pub fn new(
         }
     });
 
+    // Get a basis for the system time, so that we can use it to compute
+    // the "passage of time", which is relevant for inherent data provider,
+    // when returning a timestamp relative to a custom genesis timestamp.
+    let genesis_timestamp_prereq = anvil_config
+        .genesis_timestamp
```

Comment:
let's use `get_genesis_timestamp()`, which also takes into account the genesis json file.
also, let's create a separate function for the create_inherent_data_providers instead of the closure